pkg/spoke/resource/mapper.go

```diff
@@ -35,19 +36,30 @@ func (p *Mapper) Run(stopCh <-chan struct{}) {
 	go wait.Until(func() {
 		p.syncLock.Lock()
 		defer p.syncLock.Unlock()
-		deferredMappd := p.Mapper.(*restmapper.DeferredDiscoveryRESTMapper)
+		deferredMappd := p.StandbyMapper.(*restmapper.DeferredDiscoveryRESTMapper)
 		deferredMappd.Reset()
+		// swap two mapper if the standby mapper is reset and ready to serve
+		if ready(p.StandbyMapper) {
+			p.ServingMapper, p.StandbyMapper = p.StandbyMapper, p.ServingMapper
```

Comment:
I think we could put lock within this if?

Reply:
Yes, you are right. We can do it for only `p.ServingMapper` is accessed in different Goroutines.